internal/rbd/rbd_util.go

```diff
@@ -1042,33 +1042,14 @@ func genSnapFromSnapID(
 		}
 	}
 
-	err = updateSnapshotDetails(rbdSnap)
+	err = rbdSnap.getImageInfo()
 	if err != nil {
 		return rbdSnap, fmt.Errorf("failed to update snapshot details for %q: %w", rbdSnap, err)
 	}
 
 	return rbdSnap, err
 }
 
-// updateSnapshotDetails will copy the details from the rbdVolume to the
-// rbdSnapshot. example copying size from rbdVolume to rbdSnapshot.
-func updateSnapshotDetails(rbdSnap *rbdSnapshot) error {
-	vol := rbdSnap.toVolume()
```

Comment:
this call is missing above?

Reply:
It really should not be needed, because `rbdSnapshot` is an `rbdImage` that implements the `getImageInfo()` function...